mod/blog/actions/blog/auto_save_revision.php

```diff
@@ -16,78 +16,62 @@
 	$excerpt = $description;
 }
 
-// store errors to pass along
-$error = FALSE;
-
-if ($title && $description) {
-
-	if ($guid) {
-		$entity = get_entity($guid);
-		if (elgg_instanceof($entity, 'object', 'blog') && $entity->canEdit()) {
-			$blog = $entity;
-		} else {
-			$error = elgg_echo('blog:error:post_not_found');
-		}
-	} else {
-		$blog = new ElggBlog();
-		$blog->subtype = 'blog';
+if (!$title || !$description) {
+	return elgg_error_response(elgg_echo('blog:error:missing:description'));
+}
 
-		// force draft and private for autosaves.
-		$blog->status = 'unsaved_draft';
-		$blog->access_id = ACCESS_PRIVATE;
-		$blog->title = $title;
-		$blog->description = $description;
-		$blog->excerpt = elgg_get_excerpt($excerpt);
+if ($guid) {
+	$blog = get_entity($guid);
+	if (!$blog instanceof ElggBlog || !$blog->canEdit()) {
+		return elgg_error_response(elgg_echo('blog:error:post_not_found'));
+	}
+} else {
+	$blog = new ElggBlog();
+	$blog->subtype = 'blog';
 
-		// mark this as a brand new post so we can work out the
-		// river / revision logic in the real save action.
-		$blog->new_post = TRUE;
+	// force draft and private for autosaves.
+	$blog->status = ElggBlog::UNSAVED_DRAFT;
+	$blog->access_id = ACCESS_PRIVATE;
+	$blog->title = $title;
+	$blog->description = $description;
+	$blog->excerpt = elgg_get_excerpt($excerpt);
 
-		if (!$blog->save()) {
-			$error = elgg_echo('blog:error:cannot_save');
-		}
+	if (!$blog->save()) {
```

Comment:
It's bad that we allow blogs to be saved even though the container may change in the save action and prevent the blog from being saved altogether...